cmd/soroban-cli/src/commands/keys/generate.rs

```diff
@@ -60,17 +100,36 @@ impl Cmd {
         } else {
             seed_phrase
         };
-        self.config_locator.write_identity(&self.name, &secret)?;
-        if !self.no_fund {
-            let addr = secret.public_key(self.hd_path)?;
-            let network = self.network.get(&self.config_locator)?;
-            network
-                .fund_address(&addr)
-                .await
-                .map_err(|e| {
-                    tracing::warn!("fund_address failed: {e}");
-                })
-                .unwrap_or_default();
+
+        // Check if identity exists
+        match self.config_locator.read_identity(&self.name) {
+            Ok(existing_secret) => {
+                if self.seed.is_some() || self.default_seed {
+                    // Compare secrets only if seed is provided
+                    match (
+                        existing_secret.private_key(self.hd_path),
+                        secret.private_key(self.hd_path),
+                    ) {
+                        (Ok(existing_pk), Ok(new_pk)) if existing_pk == new_pk => {
+                            self.handle_existing_identity(&existing_secret).await?;
+                            return Ok(());
+                        }
+                        _ => {
+                            // Secrets don't match
+                            eprintln!("An identity with the name {} already exists but has a different secret. Overwriting...", self.name);
+                            self.write_and_fund_identity(&secret).await?;
+                        }
+                    }
+                } else {
+                    // No seed provided, inform user that identity already exists
+                    self.handle_existing_identity(&existing_secret).await?;
+                    return Ok(());
+                }
+            }
+            Err(_) => {
+                // Identity doesn't exist, create new one
+                self.write_and_fund_identity(&secret).await?;
+            }
```

Comment:
Too hard to write inline suggestion, but I would use the following since you don't care what the error is.
```suggestion
        let Some(existing_secret) = self.config_locator.read_identity(&self.name) {
        ...
        } else {
             // Identity doesn't exist, create new one
            self.write_and_fund_identity(&secret).await?;
        };
    
```